core/src/main/java/fj/data/TreeMap.java

```diff
@@ -295,4 +295,20 @@ public boolean contains(final K k) {
     return new TreeMap<K, W>(tree.map(TreeMap.<K, Option<W>>ord(o), g));
   }
 
+  /**
+   * Extend this TreeMap to include all mapping from another TreeMap.  Where a
+   * mapping exists in both this TreeMap and the other one, the value from the
+   * other one will replace the value in this one.
+   *
+   * @param other
+   * @return
+   */
+	public Object setAll(TreeMap<K, V> other) {
+		TreeMap<K, V> result = this;
+		for(P2<K,V> p : other) {
+			result = result.set(p._1(), p._2());
+		}
+		return result;
+	}
+
```

Comment:
For later reference, see "Implementing Sets Efficiently in a Functional Language" by Stephen Adams as mentioned here, https://github.com/purescript/purescript-maps/issues/1.

A current link to the paper is:

http://citeseerx.ist.psu.edu/viewdoc/download;jsessionid=77314EAFB757E19867DA32911F66400C?doi=10.1.1.48.1134&rep=rep1&type=pdf

Implementation in Haskell:

https://downloads.haskell.org/~ghc/7.6.2/docs/html/libraries/containers-0.5.0.0/src/Data-Set-Base.html#hedgeUnion